src/pages/index.js

```diff
@@ -17,47 +47,138 @@ popupImage.setEventListeners();
 const openPopupImage = (data) => {
   popupImage.open(data);
 };
-const renderCard = (card) => {
-  cardRenderer.addItem(createCard(card));
+
+const deletePopup = new PopupWithForm(constants.deletePopup, (data) => {
+  deletePopup.renderLoading(true);
+  api
+    .deleteCard(data._id, data)
+    .then(() => {
+      const card = document.getElementById(data._id);
+
+      card.remove();
+
+      deletePopup.close();
+    })
+
+    .catch(console.error)
+
+    .finally(() => {
+      deletePopup.renderLoading(false, "Delete Card");
+    });
+});
+deletePopup.setEventListeners();
+
+const openDeletePopup = (data) => {
+  deletePopup.open();
+  constants.deleteID.value = data._id;
 };
-const cardRenderer = new Section(
-  { items: constants.initialCards, renderer: renderCard },
-  constants.cardList
-);
 
-cardRenderer.renderItems();
+function sendLike(data) {
+  if (data.isLiked) {
+    api
+      .removeLike(data._id, data)
+      .then((result) => {
+        data.isLiked = result.isLiked;
+      })
+      .catch(console.error);
+  } else {
+    api
+      .likeCard(data._id, data)
+      .then((result) => {
+        data.isLiked = result.isLiked;
+      })
+      .catch(console.error);
+  }
+}
 
 const userInfo = new UserInfo({
   name: constants.profileName,
-  description: constants.profileDesc,
+  about: constants.profileDesc,
+  avatar: constants.avatar,
 });
 const cardModal = new PopupWithForm(constants.cardModal, (data) => {
-  cardRenderer.addItem(createCard(data));
+  cardModal.renderLoading(true);
+  api
+    .sendCard(data)
+
+    .then((result) => {
+      cardSection.addItem(createCard(result));
+
+      cardModal.close();
+    })
+    .catch(console.error)
+    .finally(() => {
+      cardModal.renderLoading(false, "Create");
+    });
 });
+
 cardModal.setEventListeners();
+const avatarModal = new PopupWithForm(constants.changeAvatar, (data) => {
+  avatarModal.renderLoading(true);
+  api
+    .updateProfilePicture(data)
+    .then((result) => {
+      constants.avatar.src = result.avatar;
```

Comment:
`      constants.avatar.src = result.avatar;
`

Class `UserInfo` should set the entire user’s data (including the **avatar**)



```js

userInfo.setUserInfo(result);
```